src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -368,29 +338,33 @@ private static object BindToCollection(Type type, IConfiguration config, BinderO
                         return;
                     }
 
-                    object? boundFromInterface = AttemptBindToCollectionInterfaces(type, config, options);
-                    if (boundFromInterface != null)
+                    (bool wasInterface, object? instance) = AttemptBindToCollectionInterfaces(type, config, options);
+                    if (wasInterface)
                     {
-                        bindingPoint.SetValue(boundFromInterface);
+                        if (instance != null)
+                        {
+                            bindingPoint.SetValue(instance);
+                        }
```

Comment:
I think hiding this logic in `AttemptBindToCollectionInterfaces` obfuscates what's really happening.

I checked out your branch because this logic is really complicated and I wanted to make sure my suggestions didn't break any important scenarios. 

Here are the follow up changes I would make to simplify this logic.

https://github.com/dotnet/runtime/compare/fcddddc80204ac920ee23d06f0af84e212792237...halter73/configbinder-iset

While doing this I realized that the more consistent behavior is to throw for null properties with a custom interface type, so I changed the behavior and updated the test.

Reply:
Thanks for help. I've copied over your changes.